src/socket.rs

```diff
@@ -240,6 +277,51 @@ impl Sockets {
         self.v6.as_ref().map(Arc::clone)
     }
 
+    /// Add a new IPv4 interface for multicast operations.
+    /// Returns Ok(()) if the socket was successfully created and added.
+    pub fn add_interface_v4(&self, addr: Ipv4Addr) -> Result<(), SocketError> {
+        // Create the interface-specific socket for sending
+        let socket = socket_v4(Some(addr))?;
+
+        let mut interfaces = self.interface_sockets_v4.write().unwrap();
+        // Check if interface already exists
+        if !interfaces.contains_key(&addr) {
+            interfaces.insert(addr, Arc::new(socket));
+            tracing::info!("Added interface {} for multicast", addr);
+        }
+        Ok(())
+    }
+
+    /// Remove an IPv4 interface from multicast operations.
+    /// Returns true if the interface was found and removed.
+    pub fn remove_interface_v4(&self, addr: Ipv4Addr) -> bool {
+        let mut interfaces = self.interface_sockets_v4.write().unwrap();
+
+        if interfaces.contains_key(&addr) {
+            interfaces.remove(&addr);
```

Comment:
same here as above, but the other way around: dropping the socket could block for some time, right? so better release the lock before dropping the socket